src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationSource.cs

```diff
@@ -32,6 +33,11 @@ public abstract class FileConfigurationSource : IConfigurationSource
         public bool ReloadOnChange { get; set; }
 
         /// <summary>
+        /// Will be called if an uncaught exception occurs in FileConfigurationProvider.Load.
+        /// </summary>
+        public Action<Exception> OnLoadError { get; set; }
```

Comment:
I like the idea, although not sure the meaning of returning true vs. false is clear. Maybe passing a "load context" object that contains the exception, the sender (provider) and a bool to decide the outcome?